recipes/newrelic/apm/dotNet/debian-systemd-nginx.yml

```diff
@@ -0,0 +1,283 @@
+              read answer
+              echo ""
+              NEW_RELIC_CONTINUE=$(echo "${answer^^}" | cut -c1-1)
+              if [[ -z "$NEW_RELIC_CONTINUE" ]]; then
+                NEW_RELIC_CONTINUE="Y"
+              fi
+              if [[ "$NEW_RELIC_CONTINUE" == "N" ]]; then
+                echo "Exiting the installation"
+                exit 130
+              fi
+              if [[ "$NEW_RELIC_CONTINUE" == "Y" ]]; then
+                break
+              fi
+              echo -e "Please type Y or N only."
+            done
+          fi
+
+    assert_pre_req:
+      cmds:
+        - |
+          IS_UNZIP_INSTALLED=$(which unzip | wc -l)
+          if [ $IS_UNZIP_INSTALLED -eq 0 ]
+          then
+            echo "unzip is required to run the newrelic install. Please install unzip and re-run the installation." >> /dev/stderr
+            exit 17
+          fi
+
+    introspector:
+      cmds:
+        - rm -rf /tmp/dotnet-introspector
+        - curl -s https://virtuoso-testing.s3-us-west-2.amazonaws.com/dotnet-is-linux-x64.zip -o ~/dotnet-is-linux-x64.zip
+        - mkdir -p /tmp/dotnet-introspector
+        - unzip -o ~/dotnet-is-linux-x64.zip -d /tmp/dotnet-introspector > /dev/null
+        - chmod -R 777 /tmp/dotnet-introspector
+        # nri-lsi-dotnet output looks like "[18044,26844,38720,31448]"
+        - /tmp/dotnet-introspector/nri-lsi-dotnet -c | tr -d '[]' | tr ',' '\n' > /tmp/dotnet-introspector/dotnet_processes.txt
+        - |
+          processCount=$(cat /tmp/dotnet-introspector/dotnet_processes.txt | wc -w)
+          if [[ "$processCount" -eq 0 ]]; then
+            YELLOW='\033[0;33m'
+            NOCOLOR='\033[0m'
+            echo -e "${YELLOW}
+            No dotnet processes were found.  Please ensure the services you want to instrument
+            are running and try running this installer again.
+            ${NOCOLOR}\n"
+            exit 3
+          fi
+        - echo "Done with instrospector"
+
+    gather_services:
+      cmds:
+        - systemctl list-unit-files --type service | awk '{ print $1 }' > /tmp/dotnet-introspector/all_services.txt
+        - cat /dev/null > /tmp/dotnet-introspector/dotnet_services.txt
+        - |
+          for sn in `cat /tmp/dotnet-introspector/all_services.txt`; do
+              mainpid=$(systemctl show --property MainPID $sn 2> /dev/null | sed 's/MainPID=//')
+              for dnp in `cat /tmp/dotnet-introspector/dotnet_processes.txt`; do
+                  if [[ "$mainpid" -eq "$dnp" ]]
+                  then
+                    echo Found .NET service: $sn
+                    echo $sn >> /tmp/dotnet-introspector/dotnet_services.txt
+                  fi
+              done
+          done
+        - echo "Done gathering services"
+
+    stop:
+      cmds:
+        - |
+          for sn in `cat /tmp/dotnet-introspector/dotnet_services.txt`; do
+            sudo systemctl stop $sn
```

Comment:
We might need to handle other system command than just systemctl?
Here is an example https://github.com/newrelic/open-install-library/blob/main/recipes/newrelic/infrastructure/debian.yml#L185-L201

Reply:
I think the scope for the current work with .NET is only to support systemctl.  We can expand the support in the future.  I do think we would ensure that systemctl is on the host before getting to far into the recipe.  We did this in the Node recipes:
https://github.com/newrelic/open-install-library/blob/main/recipes/newrelic/apm/node/linux.yml#L49

Reply:
Assuming we can target the recipe to linuxes we know have a .net core app, this might not be too much of an issue.
Yes, we would have to add some precheck assertions to throw a well known error code if we don't see systemctl is available and running.

Reply:
I've added a step to make sure `systemctl` exists to our `assert_pre_reqs` step for this iteration of the recipe.

Reply:
True, but I think that would be future work or maybe even another recipe.